guides/dimensiondata.md

```diff
@@ -0,0 +1,721 @@
+
+The following example shows you how to create a new server in the virtual datacenter created above:
+{% highlight java %}
+        /*
+         * The Server that gets deployed will have some network configuration. It gets assigned to the Vlan that was created previously.
+         */
+        NetworkInfo networkInfo = NetworkInfo
+                .create(networkDomainId, NIC.builder().vlanId(vlanId).build(), Lists.<NIC>newArrayList());
+        /*
+         * The Server that gets deployed will have some additional disk configuration.
+         */
+        List<Disk> disks = ImmutableList.of(Disk.builder().scsiId(0).speed("STANDARD").build());
+
+        /*
+         * The Server is deployed using the OS Image we selected,
+         * a flag to signal if we want it started or not, an admin pass and the additional configuration we built.
+         */
+        String serverId = api.getServerApi()
+                .deployServer("jclouds-server", imageId, true, networkInfo, "P$$ssWwrrdGoDd!", disks, null);
+
+{% endhighlight %}
+
+ A Server deployment is an asynchronous process. We need to wait for it to complete. The Dimension Data provider has built in predicates that will block execution and check that the Server's State has moved from PENDING_ADD to NORMAL. This piece of sample code also waits for the Server to be started.
+{% highlight java %}
+    static void waitForServerStartedAndNormal(Injector injector, String serverId)
+    {
+        Predicate<String> serverStartedPredicate = injector.getInstance(Key.get(new TypeLiteral<Predicate<String>>()
+        {
+        }, Names.named("SERVER_STARTED_PREDICATE")));
+        Predicate<String> serverNormalPredicate = injector.getInstance(Key.get(new TypeLiteral<Predicate<String>>()
+        {
+        }, Names.named("SERVER_NORMAL_PREDICATE")));
+
+        // Wait for Server to be started and NORMAL
+        serverStartedPredicate.apply(serverId);
+        serverNormalPredicate.apply(serverId);
+    }
+{% endhighlight %}
+
+## <a id="how-to-apply-a-tag-to-an-asset"></a>How to: Apply a Tag to an Asset
+
+Dimension Data CloudControl allows for Tags to be applied to assets. There are two parts to the tagging process, first a tag key needs to be created, second the tag key is applied to an asset. 
+
+Create a tag key. We will use this to tag the assets that we create.
+{% highlight java %}
+        String tagKeyId = api.getTagApi().createTagKey("jclouds", "owner of the asset", true, false);
+{% endhighlight %}
+
+Follows is the sample code for applying a Tag to the Server. We use AssetType SERVER. Pass in the tagKeyId and a value that we want to associate, in this case jclouds.
+{% highlight java %}
+        api.getTagApi().applyTags(serverId, "SERVER", Collections.singletonList(TagInfo.create(tagKeyId, "jclouds")));
+{% endhighlight %}
+
+The full set of AssetTypes are SERVER, NETWORK_DOMAIN, VLAN, CUSTOMER_IMAGE, PUBLIC_IP_BLOCK or ACCOUNT.
+
+## <a id="examples"></a>Examples:
```

Comment:
This is gonna become obsolete quickly. Remove the code examples here and link to the examples repo.

Reply:
I have updated with the assumed links to master (when the https://github.com/jclouds/jclouds-examples/pull/94 is merged)

Reply:
Instead of doing this, I'd just add one line saying that further and more detailed examples can be found in the examples repo under the `dimensiondata` folder. Just one line. Otherwise, are we going to add every example we may add here? Let's just add that note.